x-pack/solutions/observability/plugins/synthetics/server/tasks/sync_private_locations_monitors_task.ts

```diff
@@ -0,0 +1,288 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { TaskManagerSetupContract } from '@kbn/task-manager-plugin/server/plugin';
+import {
+  SavedObjectsClientContract,
+  SavedObjectsFindResult,
+} from '@kbn/core-saved-objects-api-server';
+import { EncryptedSavedObjectsPluginStart } from '@kbn/encrypted-saved-objects-plugin/server';
+import { ALL_SPACES_ID } from '@kbn/spaces-plugin/common/constants';
+import { RunContext } from '@kbn/task-manager-plugin/server';
+import moment from 'moment';
+import { syntheticsParamType } from '../../common/types/saved_objects';
+import { normalizeSecrets } from '../synthetics_service/utils';
+import type { PrivateLocationAttributes } from '../runtime_types/private_locations';
+import {
+  HeartbeatConfig,
+  MonitorFields,
+  SyntheticsMonitorWithSecretsAttributes,
+} from '../../common/runtime_types';
+import { MonitorConfigRepository } from '../services/monitor_config_repository';
+import { SyntheticsMonitorClient } from '../synthetics_service/synthetics_monitor/synthetics_monitor_client';
+import { getPrivateLocations } from '../synthetics_service/get_private_locations';
+import { SyntheticsServerSetup } from '../types';
+import {
+  formatHeartbeatRequest,
+  mixParamsWithGlobalParams,
+} from '../synthetics_service/formatters/public_formatters/format_configs';
+
+const TASK_TYPE = 'Synthetics:Sync-Private-Location-Monitors';
+const VERSION = '1.0.0';
+const TASK_ID = `${TASK_TYPE}:${VERSION}`;
+
+export class SyncPrivateLocationMonitorsTask {
+  constructor(
+    public serverSetup: SyntheticsServerSetup,
+    public taskManager: TaskManagerSetupContract,
+    public syntheticsMonitorClient: SyntheticsMonitorClient
+  ) {
+    taskManager.registerTaskDefinitions({
+      [TASK_TYPE]: {
+        title: 'Synthetics Sync Global Params Task',
+        description:
+          'This task is executed so that we can sync private location monitors for example when global params are updated',
+        timeout: '3m',
+        maxAttempts: 3,
+        createTaskRunner: (taskInstance) => {
+          return {
+            run: async () => {
```

Comment:
You should probably handle `cancel` as well as run; from a brief look at what the task is doing, it looks like it's iterating through spaces, and we've had customers with 1000's of spaces ...

The best approach to handling cancel seems to be to create an AbortController, that would get passed to the executor, and the `cancel()` method would just call `abortController.abort()`.  The executor code can pass the `abortSignal.signal` to any APIs that abort controllers (es library, for example), or check for cancellation with `abortSignal.signal.aborted`.